### PR TITLE
Add speedup ability

### PR DESCRIPTION
**Pull request by sparshg**
- ~~Removed unused variables~~
- ~~Moved `__init__`, `init_game`, `set_level` to top of class~~
- Added a method to `set_time_step`
- When down arrow key held, speedup ticks

### CODE REVIEW

**Comment by Shreyas-SAS** [comment]
> * Moved `__init__`, `init_game`, `set_level` to top of class

I could not get it, can you please explain me the reason for this.

**Comment by sparshg** [comment]
I think initializing functions should be moved to top of class, it helps to understand/debug the code better

**Comment by Shreyas-SAS** [comment]
> I think initializing functions should be moved to top of class, it helps to understand/debug the code better

oh so it was not to specifically make a change to the code but increase understanding. I got it now and really appreciate your work there.

**Comment by Shreyas-SAS** [comment]
> * Removed unused variables

I suggest keeping **`sound = off`** on line 144(new code) or 154(old code). if there was a reason removing it that i did not realize i would really love to learn about it.

**Comment by sparshg** [comment]
`sound` variable is not being used anywhere and seems redundant, `soundon` has beenused to toggle sound

**Comment by Shreyas-SAS** [comment]
> `sound` variable is not being used anywhere and seems redundant, `soundon` has been used to toggle sound

but initial declaration of `sound` as **off** serves to create a base before the toggle between `soundon`, thus fixing the initial sound to be off and removing any unwanted noise present initially. although once `soundon` is used `sound` seems to go redundant but I still suggest keeping it, as a single variable does not occupy enough space to actually make a very significant difference in resources and speed.

**Comment by quozl** [comment]
Thank you both for the chatter, it's been interesting.  However, your pull request is not yet accepted because;
* [ ] you remove a feature without good reason; the down arrow key has a purpose already,
* [ ] you mix code layout changes with feature changes in the same commit,
* [ ] the commit message does not explain the work done.

**Comment by sparshg** [comment]
Thank you for the review. I reverted the layout changes for this PR.
It seems more intuitive for the down arrow to speedup the descent rate and space to drop it.
Also down arrow key was unused, (only space key, and numpad down arrow worked to drop down block).

**Comment by quozl** [comment]
> Thank you for the review. I reverted the layout changes for this PR.

Thanks, I see you pushed revert as a new commit. Please rewrite your commits instead.  There are several ways to do this, so I won't go into detail.  Read documentation for Git.

> It seems more intuitive for the down arrow to speedup the descent rate and space to drop it.

Thanks for explaining your reasoning.  That explanation should be added to the commit message.

> Also down arrow key was unused, (only space key, and numpad down arrow worked to drop down block).

That's not strictly true.  KP_Down was used, and your patch reclaims it for another purpose.  I agree Down was not used, but that seems to have been an oversight, given how the other pairs of arrow keys were used.

For my part, I don't think this new feature warrants breaking the binding to the old feature, as an upgrade of the activity will behave differently with respect to keyboard shortcuts.  I welcome comments from other Sugar Labs members on this.

> For the commit message, I don't know how to change previous commit's message :\, would be helpful if you can guide me a bit, thanks!

Please do your own research, evaluate the many methods, and choose one that fits your workflow and tools.  I've no idea what your workflow has been to this point, so I can't recommend any specific method.  I also don't want to get into the role of explaining commonly available and well documented tools.

**Comment by sparshg** [comment]
I rewrote the commits and included a detailed commit message too. I still think down arrow to speedup descent rate is a good idea, almost all existing Tetris-like games I saw have this feature. Thanks for the review though!

**Comment by quozl** [comment]
Thanks.  While we wait for a second opinion, I have looked at your rewritten commit message.  GitHub splits it up and makes it hard to read.  See the commit view 78e78bc53b5d267faf9cc7c3224823fede3239de.  It happens because of convention to use first line separate to remaining lines.

You may be interested in our documentation for [writing commit messages](https://github.com/sugarlabs/sugar-docs/blob/master/src/contributing.md#making-commits).

**Comment by sparshg** [comment]
Thanks, I changed the commit message as per the guidelines :D

**Comment by quozl** [comment]
Thanks.  Extra space before _method_ and no period at end of sentence.  :grin:

File: BlockParty.py [hunk 1]
```diff
@@ -513,6 +524,7 @@ def init_game(self):
         self.clear_glass()
         self.complete_update = True
         self.glass_update = True
+        self.canSpeedUp = True
```
Comment (chimosky) [inline BlockParty.py:527]: Could you stick to the naming convention in the repo? You've used camel case naming and that's not what's used in the repo.

**Review by chimosky — CHANGES_REQUESTED** [review]

File: BlockParty.py (hunk 1 above)
Comment (sparshg) [reply BlockParty.py]: Oops, updated :D

**Pull request merged**